crates/evm/core/src/backend/mod.rs

```diff
@@ -1894,16 +1906,26 @@ fn commit_transaction<I: InspectorExt<Backend>>(
     };
     trace!(elapsed = ?now.elapsed(), "transacted transaction");
 
-    apply_state_changeset(res.state, journaled_state, fork)?;
+    apply_state_changeset(res.state, journaled_state, fork, persistent_accounts)?;
     Ok(())
 }
 
 /// Helper method which updates data in the state with the data from the database.
-pub fn update_state<DB: Database>(state: &mut EvmState, db: &mut DB) -> Result<(), DB::Error> {
+/// Does not change state for persistent accounts (for roll fork to transaction and transact).
+pub fn update_state<DB: Database>(
+    state: &mut EvmState,
+    db: &mut DB,
+    persistent_accounts: Option<&HashSet<Address>>,
+) -> Result<(), DB::Error> {
     for (addr, acc) in state.iter_mut() {
-        acc.info = db.basic(*addr)?.unwrap_or_default();
-        for (key, val) in acc.storage.iter_mut() {
-            val.present_value = db.storage(*addr, *key)?;
+        let is_persistent =
+            if let Some(accounts) = persistent_accounts { accounts.contains(addr) } else { false };
```

Comment:
this looks a bit strange, 
I'd prefer persistent_accounts.and_then().unwrap_or(false)

Reply:
went for `map_or` in https://github.com/foundry-rs/foundry/pull/8129/commits/777764d94b3918bda4ac70da9193337670fd190b, looks a little bit more cleaner